pkg/api/pod/util.go

```diff
@@ -428,6 +428,7 @@ func GetValidationOptionsFromPodSpecAndMeta(podSpec, oldPodSpec *api.PodSpec, po
 		OldPodViolatesLegacyMatchLabelKeysValidation:        false,
 		AllowContainerRestartPolicyRules:                    utilfeature.DefaultFeatureGate.Enabled(features.ContainerRestartRules),
 		AllowUserNamespacesWithVolumeDevices:                false,
+		AllowContainerRestartPolicyRulesOnSidecars:          utilfeature.DefaultFeatureGate.Enabled(features.RestartAllContainersOnContainerExits),
```

Comment:
is this called FooOnSidecars because we use it for sidecars? I think it is confusing more then helping.